pulsar-broker/src/test/java/org/apache/pulsar/broker/transaction/buffer/TransactionLowWaterMarkTest.java

```diff
@@ -325,4 +327,127 @@ public void testTBLowWaterMarkEndToEnd() throws Exception {
             // no-op
         }
     }
+
+    @Test
+    public void testLowWaterMarkForDifferentTC() throws Exception {
```

Comment:
why we need txn4 and txn5, I think when txn3 has committed the txn1 and txn2 will be write abort marker. and also we need to test the TopicTransactionBuffer recover and the will recover to a correct state

Reply:
We need txn3 and txn4 to update lowWaterMark of its TC.

Reply:
Maybe we should use reflection to modify the map to test this case instead of txn3 and txn4.
I will optimize it  later

Reply:
only txn 3 is enough why we need txn4

Reply:
transaction 1 has different TC with transaction 2